### PR TITLE
small fixes for tcollector

### PR DESCRIPTION
**Pull request by cacktopus**
@rf @Raynos @kriskowal

### CODE REVIEW

**Comment by kriskowal** [comment]
Looks :ok: to me.

File: node/tcollector/reporter.js [hunk 1]
```diff
@@ -112,11 +112,8 @@ function jsonSpanToThriftSpan(span) {
             return ret;
         });
 
-    if (!span.host) {
-        // Workaround to support older tchannels
-        span.host =
-            TCollectorTraceReporter.convertHost(span.annotations[0].host);
-    }
+    var endpoint = span.endpoint || span.annotations[0].host;
+    var host = TCollectorTraceReporter.convertHost(endpoint);
```
Comment (Raynos) [inline node/tcollector/reporter.js]: This looks strange; are you sure `span.endpoint` is a host ?

Comment (Raynos) [inline node/tcollector/reporter.js]: turns out it is a host..

**Comment by Raynos** [comment]
lgtm

**Pull request merged**